pulp_rpm/tests/performance/conftest.py

```diff
@@ -0,0 +1,30 @@
+
+from itertools import groupby
+from pulp_rpm.tests.performance.utils import (
+    ConsumedRamLogEntry,
+    consumed_ram_log,
+    get_consumed_ram,
+    START,
+    END,
+)
+
+
+def pytest_runtest_setup(item):
+    """Starting to measure memory on setup."""
+    log_entry = ConsumedRamLogEntry(item.nodeid, START, get_consumed_ram())
+    consumed_ram_log.append(log_entry)
+
+
+def pytest_runtest_teardown(item):
+    """Logging memory consumption on teardown."""
+    log_entry = ConsumedRamLogEntry(item.nodeid, END, get_consumed_ram())
+    consumed_ram_log.append(log_entry)
+
+
+def pytest_terminal_summary(terminalreporter):
+    """Display memory consumption on terminal."""
+    grouped = groupby(consumed_ram_log, lambda entry: entry.nodeid)
+    for nodeid, (start_entry, end_entry) in grouped:
+        leaked = end_entry.consumed_ram - start_entry.consumed_ram
+        terminalreporter.write('Consumed Memory: {:.2f}MB in {}\n'.format(
```

Comment:
If I understand correctly, it's the amount of leaked memory. If this was the goal, maybe we should report is as "leaked", what do you think?

Memory consumption would be the amount of memory a test requires at a certain time.
I think the most useful information for us would be to know the max memory consumption for a test. E.g. the maximum amount of memory used for centos7 sync.

Let me know if I miss anything.

Reply:
I changed it to leaked, but I liked your suggestion to get the maximum amount of consumed memory, I will try to add it

Reply:
this PR already helped me to profile memory:

Reply:
this PR already helped me to find a problem:
https://pulp.plan.io/issues/5688?#note-7